presto-main/src/main/java/io/prestosql/sql/planner/LiteralEncoder.java

```diff
@@ -226,6 +230,25 @@ public Expression toExpression(Object object, Type type)
             return new TimestampLiteral(representation);
         }
 
+        if (type instanceof TimestampWithTimeZoneType) {
+            TimestampWithTimeZoneType timestampWithTimeZoneType = (TimestampWithTimeZoneType) type;
+            String representation;
+            if (timestampWithTimeZoneType.isShort()) {
+                representation = TimestampWithTimeZoneToVarcharCast.cast(timestampWithTimeZoneType.getPrecision(), (long) object).toStringUtf8();
+            }
+            else {
+                representation = TimestampWithTimeZoneToVarcharCast.cast(timestampWithTimeZoneType.getPrecision(), (LongTimestampWithTimeZone) object).toStringUtf8();
+            }
+            if (!object.equals(parseTimestampWithTimeZone(timestampWithTimeZoneType.getPrecision(), representation))) {
+                // Certain (point in time, time zone) pairs cannot be represented as a TIMESTAMP literal, as the literal uses local date/time in given time zone.
+                // Thus, during DST backwards change by e.g. 1 hour, the local time is "repeated" twice and thus one local date/time logically corresponds to two
+                // points in time, leaving one of them non-referencable.
```

Comment:
Link to https://github.com/prestosql/presto/issues/5781

Reply:
right, link added